packages/op-datasets/src/op_datasets/logic/transforms/daily_address_summary.py

```diff
@@ -0,0 +1,209 @@
+    groupby_cols = list(groupby_args.values())
+
+    # SQL query to perform aggregations
+    groupby_clause = ", ".join(groupby_cols)
+
+    query = f"""
+    WITH constants AS (
+        SELECT
+            CAST(1000000000000000000 AS DECIMAL(18, 0)) AS WEI_IN_ETH,
+            CAST(1000000000 AS DECIMAL(9, 0)) AS WEI_IN_GWEI
+    )
+    ,df_enrich AS (
+        SELECT
+            _filter_df.*
+
+            -- Gas Fee Breakdown
+            ,CAST(((gas_price * receipt_gas_used) + receipt_l1_fee) AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS total_gas_fee
+            ,CASE
+                WHEN receipt_status = 1 THEN CAST(((gas_price * receipt_gas_used) + receipt_l1_fee) AS DECIMAL(38,18)) / constants.WEI_IN_ETH
+                ELSE 0
+                END AS total_gas_fee_success
+            ,CASE
+                WHEN receipt_status != 1 THEN CAST(((gas_price * receipt_gas_used) + receipt_l1_fee) AS DECIMAL(38,18)) / constants.WEI_IN_ETH
+                ELSE 0
+                END AS total_gas_fee_fail
+            ,CAST((gas_price * receipt_gas_used) AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS l2_contrib_gas_fee
+            ,CAST(receipt_l1_fee AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS l1_contrib_gas_fee
+            ,CAST((receipt_l1_gas_used * receipt_l1_blob_base_fee_scalar * receipt_l1_blob_base_fee) AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS l1_blobgas_contrib_gas_fee
+            ,CAST((receipt_l1_gas_used * COALESCE(receipt_l1_base_fee_scalar, receipt_l1_fee_scalar) * receipt_l1_gas_price) AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS l1_l1gas_contrib_gas_fee
+            ,CAST((gas_price - max_priority_fee_per_gas) * receipt_gas_used AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS l2_contrib_gas_fee_base_fee
+            ,CAST(max_priority_fee_per_gas * receipt_gas_used AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS l2_contrib_gas_fee_priority_fee
+
+            -- Gas Price Breakdown, conversion for average calculation below
+            ,CAST(CAST((gas_price * receipt_gas_used) AS DECIMAL(38,9)) / constants.WEI_IN_GWEI AS DECIMAL(38,9)) AS total_gas_price_gwei
+            ,CAST(CAST(((gas_price - max_priority_fee_per_gas) * receipt_gas_used) AS DECIMAL(38,9)) / constants.WEI_IN_GWEI AS DECIMAL(38,9)) AS base_fee_gwei
+            ,CAST(CAST(max_priority_fee_per_gas * receipt_gas_used AS DECIMAL(38,9)) / constants.WEI_IN_GWEI AS DECIMAL(38,9)) AS priority_fee_gwei
+            ,CAST(CAST(receipt_l1_gas_price * receipt_l1_gas_used AS DECIMAL(38,9)) / constants.WEI_IN_GWEI AS DECIMAL(38,9)) AS l1_gas_price_gwei
+            ,CAST(CAST(receipt_l1_blob_base_fee * receipt_l1_gas_used AS DECIMAL(38,9)) / constants.WEI_IN_GWEI AS DECIMAL(38,9)) AS l1_blob_base_fee_gwei
+        FROM
+            _filter_df
+        CROSS JOIN constants
+    )
+
+    SELECT
+        {groupby_clause}
+
+        -- Transactions
+        ,COUNT(hash) AS total_txs
+        ,COUNT(CASE WHEN receipt_status = 1 THEN hash ELSE NULL END) AS total_txs_success
```

Comment:
case when success could be a utility function?

like: `count( if_tx_success(hash, null) )`